tempesta_fw/http.c

```diff
@@ -1162,23 +1215,38 @@ tfw_http_conn_collect(TfwSrvConn *srv_conn, struct list_head *sch_queue,
  * with minimal effort.
  */
 static void
-tfw_http_conn_resched(struct list_head *sch_queue, struct list_head *equeue)
+tfw_http_conn_resched(struct list_head *sch_queue, struct list_head *equeue,
+		      TfwSrvConn *prev_conn)
 {
 	TfwSrvConn *sch_conn;
 	TfwHttpReq *req, *tmp;
+	TfwServer *srv = (TfwServer *)prev_conn->peer;
 
 	/*
 	 * Process the complete queue and re-schedule all requests
-	 * to other servers/connections.
+	 * to other servers/connections. Health monitoring requests
+	 * must be re-scheduled to the same server (other servers
+	 * may not have enabled health monitor).
 	 */
 	list_for_each_entry_safe(req, tmp, sch_queue, fwd_list) {
-		if (!(sch_conn = tfw_sched_get_srv_conn((TfwMsg *)req))) {
+		if (req->flags & TFW_HTTP_HMONITOR) {
+			sch_conn = srv->sg->sched->sched_srv_conn((TfwMsg *)req,
+								  srv);
+			if (!sch_conn) {
+				list_del_init(&req->fwd_list);
+				tfw_http_msg_free((TfwHttpMsg *)req);
```

Comment:
So, a health monitoring request still cannot be sent to the server. We just remove it and do nothing. Just like when no health monitoring requests are sent out at all. So what's the difference? Is that the intended behaviour?

Reply:
Warning added.